controllers/v1/programsController.js

```diff
@@ -58,59 +58,504 @@ module.exports = class Programs extends Abstract {
+    "result": {
+        "_id": "5ff09aa4a43c952a32279234"
+    }
+   } 
+  * @apiUse successBody
+  * @apiUse errorBody
+  */
+
+  /**
+   * Create program.
+   * @method {POST}
+   * @name create
+   * @param {Object} req - requested data.
+   * @returns {JSON} - created program document.
+   */
+
+  async create(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        req.body.userId = req.userDetails.userId;
+        let programCreationData = await programsHelper.create(
+          req.body,
+          true, // checkDate
+          req.userDetails.userToken,
+        );
+
+        return resolve({
+          message: messageConstants.apiResponses.PROGRAMS_CREATED,
+          result: _.pick(programCreationData, ['_id']),
+        });
+      } catch (error) {
+        reject({
+          status: error.status || httpStatusCode.internal_server_error.status,
+          message: error.message || httpStatusCode.internal_server_error.message,
+          errorObject: error,
+        });
+      }
+    });
+  }
+
+  /**
+  * @api {post} /samiksha/v1/programs/update/:programId Update Program
+  * @apiVersion 1.0.0
+  * @apiName Update
+  * @apiGroup Programs
+  * @apiSampleRequest /samiksha/v1/programs/update/5ff09aa4a43c952a32279234
+  * @apiHeader {String} X-authenticated-user-token Authenticity token 
+  * @apiParamExample {json} Request-Body:
+  * {
+      "externalId" : "PROGID01",
+      "name" : "DCPCR School Development Index 2018-19",
+      "description" : "DCPCR School Development Index 2018-19",
+      "isDeleted" : false,
+      "resourceType" : [ 
+          "program"
+      ],
+      "language" : [ 
+          "English"
+      ],
+      "keywords" : [],
+      "concepts" : [],
+      "userId":"a082787f-8f8f-42f2-a706-35457ca6f1fd",
+      "imageCompression" : {
+          "quality" : 10
+      },
+      "components" : [ 
+          "5b98fa069f664f7e1ae7498c"
+      ],
+      "scope" : {
+          "entityType" : "state",
+          "entities" : ["5d6609ef81a57a6173a79e78"],
+          "roles" : ["HM"]
+      }
+      "startDate" : "2023-04-06T09:35:00.000Z",
+      "endDate" : ""2024-04-06T09:35:00.000Z" // optional
+    }
+  * @apiParamExample {json} Response:
+  {
+    "message": "Program updated successfully",
+    "status": 200,
+    "result": {
+        "_id": "5ff09aa4a43c952a32279234"
+    }
+   } 
+  * @apiUse successBody
+  * @apiUse errorBody
+  */
+
+  /**
+   * Update program.
+   * @method
+   * @name update
+   * @param {Object} req - requested data.
+   * @param {Object}
+   * @returns {JSON} -
+   */
+  /**
+   * Update program.
+   * @method
+   * @name update
+   * @param {Object} req - requested data.
+   * @param {Object}
+   * @returns {JSON} -
+   */
+
+  async update(req) {
+    try {
+      let programUpdationData = await programsHelper.update(
+        req.params._id,
+        req.body,
+        req.userDetails.userId,
+        true, //checkDate
+        req.userDetails.userToken,
```

Comment:
@praveenKDass token is not required, please check once

Reply:
removed token @VISHNUDAS-tunerlabs